core/src/components/button/button.ionic.scss

```diff
@@ -59,39 +59,55 @@
 :host(.button-outline) {
   --border-width: #{globals.$ion-border-size-025};
   --border-style: #{globals.$ion-border-style-solid};
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: var(--ion-color-subtle-shade, #{globals.current-color(shade, $subtle: true)});
   --background-focused: transparent;
-  --background-hover: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-hover: var(--ion-color-subtle-shade, #{globals.current-color(shade, $subtle: true)});
   --background-hover-opacity: 1;
-  --border-color: #{globals.ion-color(primary, base)};
-  --color: #{globals.ion-color(primary, base)};
+  --border-color: var(--ion-color-base, #{globals.ion-color(primary, base)});
+  --color: var(--ion-color-base, #{globals.ion-color(primary, base)});
   --ripple-opacity: var(--background-activated-opacity, 1);
   --ripple-color: var(--background-activated);
 }
 
+// Provide fallbacks for ion-color-base for all colors
+:host(.button-outline.ion-color.ion-color-warning) {
+  .button-native {
+    --ion-color-base: #{globals.$ion-text-warning};
+  }
+  ion-ripple-effect {
+    --ion-color-base: var(--ion-color-base, #{globals.current-color(shade, $subtle: true)});
+  }
+}
```

Comment:
Why is this being added? We shouldn't need fallbacks as `current-color` and `ion-color` provides fallbacks.